shared/src/bigint/bigint_ser.rs

```diff
@@ -53,5 +59,58 @@ where
             ));
         }
     };
+
+    if bz.len() > MAX_BIGINT_SIZE {
+        return Err(<D::Error as serde::de::Error>::custom("BigInt too large"));
+    }
+
     Ok(BigInt::from_bytes_be(sign, &bz[1..]))
 }
+
+#[cfg(test)]
+mod tests {
+    use fvm_ipld_encoding::{from_slice, to_vec};
+
+    use super::*;
+
+    #[test]
+    fn test_bigiint_max() {
+        let max_limbs = MAX_BIGINT_SIZE / 4; // u32 -> u8
```

Comment:
It took me a while to get the comment. Proposal for another comment: "// 32-bit limbs"

Reply:
updaetd the comment